service/src/main/java/com/codeforcommunity/processor/ProtectedReportProcessorImpl.java

```diff
@@ -54,18 +111,53 @@ public GetAdoptionReportResponse getAdoptionReport(JWTData userData) {
                 .on(ADOPTED_SITES.SITE_ID.eq(STEWARDSHIP.SITE_ID))
                 .leftJoin(NEIGHBORHOODS)
                 .on(SITES.NEIGHBORHOOD_ID.eq(NEIGHBORHOODS.ID))
+                .where(ADOPTED_SITES.DATE_ADOPTED.ge(startDate))
+                .or(TimeUnit.MILLISECONDS.toDays(startDate.getTime()) == 0)
```

Comment:
❓ Do you still need the `.or`? The stewardship query doesn't have it and passing `Date(0)` should work as expected, right?

Reply:
Without the `or`, the non-CSV route and the CSV route when the `previousDays` parameter is not given (in other words, cases where we want to get *all* adopted sites) don't work as expected for me. In these cases, sites whose `date_adopted` values are null aren't included in the response. This is because, for sites with null `date_adopted` values, the `where` clause on line 114 becomes `where(null.ge(startDate))` and this is always false, regardless of `startDate`.

The reason why the stewardship query doesn't have the `or` is because there seems to be a constraint that the `performed_on` value of a stewardship activity can't be null.

Reply:
Forgot that some of the values were null. This is good, but we should consider updating the data with the dates of the weeks the trees were adopted.

Reply:
Do you mean updating the adopted sites table so that the `DATE_ADOPTED` field can't be null like how it's done with the stewardship table (and maybe backfill some of the adopted sites whose `DATE_ADOPTED` field is already null)?